carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -1449,53 +1521,65 @@ void WMBroadcaster::addGeofence(std::shared_ptr<Geofence> gf_ptr)
   ROS_INFO_STREAM("Adding active geofence to the map with geofence id: " << gf_ptr->id_);
   
   // if applying workzone geometry geofence, utilize workzone chache to create one 
+  // also multiple map updates can be sent from one geofence object
+  std::vector<std::shared_ptr<Geofence>> updates_to_send;
+
   bool detected_workzone_signal = gf_ptr->msg_.package.label_exists && gf_ptr->msg_.package.label.find("SIG_WZ") != std::string::npos;
+  bool detected_map_msg_signal = gf_ptr->msg_.package.label_exists && gf_ptr->msg_.package.label.find("MAP_MSG") != std::string::npos;
   if (detected_workzone_signal && gf_ptr->msg_.params.detail.choice != cav_msgs::TrafficControlDetail::MAXSPEED_CHOICE)
   {
     for (auto gf_cache_ptr : work_zone_geofence_cache_)
     {
       geofenceFromMsg(gf_cache_ptr.second, gf_cache_ptr.second->msg_);
     }
-    gf_ptr = createWorkzoneGeofence(work_zone_geofence_cache_);
+    updates_to_send.push_back(createWorkzoneGeofence(work_zone_geofence_cache_));
```

Comment:
Can you add an enhancement issue to unify the workzone approach and the intersection manager? At the moment a workzone with a map message seems like it would cause some issue.

Reply:
Gotcha.